roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -0,0 +1,161 @@
+        openshift_release = "v3.4"
+        openshift_image_tag = "3.4.1.7"
+
+        is_containerized = get_var(task_vars, "openshift", "common", "is_containerized")
+
+        if is_containerized:
+            docker_or_rpm_images = set(self.containerized_docker_images(image_base_name, openshift_release))
+        else:
+            docker_or_rpm_images = set(self.rpm_docker_images(image_base_name, openshift_release))
+
+        # append images with qualified image tags to our list of required images.
+        # these are images with a (v0.0.0.0) tag, rather than a standard release
+        # format tag (v0.0). We want to check this set in both containerized and
+        # non-containerized installations.
+        docker_or_rpm_images.update(self.qualified_docker_images(self.image_from_base_name(image_base_name),
+                                                                 "v" + openshift_image_tag))
+
+        return docker_or_rpm_images
+
+    def local_images(self, required_images, task_vars):
+        """ Return all found local docker images """
+        found = set()
+        for image in required_images:
+            cmd = self.module_executor("docker_image_facts", {"name": image}, task_vars)
+            if cmd.get("images", None):
+                found.add(image)
+
+        return found
+
+    def known_docker_registries(self, task_vars):
+        docker_info = self.module_executor("docker_info", {}, task_vars).get("info", "")
+        return [registry.get("Name", "") for registry in docker_info.get("Registries", {})]
+
+    def available_images(self, missing_images, registries, task_vars):
+        """ Inspect existing images using Skopeo and return all images successfully inspected """
+        missing_images_copy = list(missing_images)
+        registries_copy = list(registries)
+
+        skopeo_update = self.update_skopeo_image(task_vars)
+        changed = skopeo_update.get("changed", False)
+
+        # exit early if Skopeo update fails
+        if skopeo_update.get("failed", False):
+            return missing_images, changed
+
+        while registries_copy and missing_images_copy:
+            registry = registries_copy.pop(0)
+            # pylint: disable=redefined-variable-type
+            missing_images_copy = self.missing_skopeo_images(registry, missing_images_copy, task_vars)
+
+        return set(missing_images) - set(missing_images_copy), changed
+
+    def containerized_docker_images(self, base_name, version):
+        return [
+            "{image}:{version}".format(image=self.image_from_base_name(base_name), version=version)
+        ]
+
+    @staticmethod
+    def rpm_docker_images(base, version):
+        return [
+            "{image_repo}/registry-console:{version}".format(image_repo=base["repo"], version=version)
+        ]
+
+    @staticmethod
+    def qualified_docker_images(image_name, version):
+        # pylint: disable=unused-argument
+        # pylint: disable=unused-variable
```

Comment:
Why is this? Ah, seems that's because of our `locals()` trick...

Perhaps it's better to rewrite the code in a simpler form:

```python
    @staticmethod
    def qualified_docker_images(image_name, version):
        return [
            "{}-{}:{}".format(image_name, component, version)
            for component in "haproxy-router docker-registry deployer pod".split()
        ]
```

---

FYI, whenever it will be needed, you can disable multiple rules in one comment:

```python
# pylint: disable=unused-argument,unused-variable
```

Our guidelines in this repo is to document why each rule is being disabled:

https://github.com/openshift/openshift-ansible/blob/master/docs/best_practices_guide.adoc#All-PyLint-rule-disables-MUST-be-documented-in-the-code (though we don't follow a consistent style in terms of how to write the justification)